### PR TITLE
Lets you empty sandbags

### PR DESCRIPTION
**Pull request by Arturlang**
## About The Pull Request
Lets you empty full sandbags back into empty sandbags by right clicking

## Why It's Good For The Game
It's wierd that this is not possible, now it's possible to do

## Changelog

:cl:
add: You can now right click full sandbags to empty them,
/:cl:

### CODE REVIEW

File: code/game/objects/items/stacks/sandbags.dm [hunk 1]
```diff
@@ -83,10 +83,39 @@
...
 	amount = 25
 
 /obj/item/stack/sandbags/attack_self(mob/living/user)
 	. = ..()
 	var/building_time = LERP(2 SECONDS, 1 SECONDS, user.skills.getPercent(SKILL_CONSTRUCTION, SKILL_ENGINEER_EXPERT))
 	create_object(user, new/datum/stack_recipe("sandbag barricade", /obj/structure/barricade/sandbags, 5, time = building_time, crafting_flags = CRAFT_CHECK_DENSITY | CRAFT_CHECK_DIRECTION | CRAFT_ON_SOLID_GROUND), 1)
+
+/obj/item/stack/sandbags/attack_self_alternate(mob/user)
+	. = ..()
+	if(get_amount() < 1)
```
Comment (Arturlang) [inline code/game/objects/items/stacks/sandbags.dm:100]: Not sure if this part is neccesary with the while loop checking it?

**Comment by DrAmazing343** [comment]
Very appreciated! I jenga my loadouts so hard and being unable to dump out bags makes it annoying because there's now two "types" to stack around :P

File: code/game/objects/items/stacks/sandbags.dm [hunk 2]
```diff
@@ -83,10 +83,39 @@
...
+			to_chat(user, span_notice("You finish emptying [src]."))
+		// check if we can stuff it into the user's hands
+		use(1)
+		var/obj/item/stack/sandbag = user.get_inactive_held_item()
+		if(istype(sandbag, /obj/item/stack/sandbags_empty) && sandbag.add(1))
+			continue
+		var/obj/item/stack/sandbags_empty/E = new(get_turf(user))
+		if(!sandbag && user.put_in_hands(E))
+			continue
+		E.add_to_stacks(user)
```
Comment (TiviPlus) [inline code/game/objects/items/stacks/sandbags.dm:121]: balloon alerts pls

File: code/game/objects/items/stacks/sandbags.dm [hunk 3]
```diff
@@ -83,10 +83,39 @@
...
+		return
+
+	to_chat(user, span_notice("You start emptying [src]."))	
+	while(get_amount() > 0)
+		if(!do_after(user, 0.5 SECONDS, IGNORE_USER_LOC_CHANGE|IGNORE_TARGET_LOC_CHANGE, user))
+			to_chat(user, span_notice("You stop emptying [src]."))
+			break
+		if(zero_amount())
+			to_chat(user, span_notice("You finish emptying [src]."))
+		// check if we can stuff it into the user's hands
```
Comment (TiviPlus) [inline code/game/objects/items/stacks/sandbags.dm:113]: shouldnt this break?

**Comment by Arturlang** [comment]
@TiviPlus did it

**Pull request merged**